packages/core/src/sync/transport.ts

```diff
@@ -45,7 +60,126 @@ export const cachedTransport = ({
       async request({ method, params }) {
         const body = { method, params };
 
+        // multicall
         if (
+          method === "eth_call" &&
+          params[0]?.data?.startsWith(MULTICALL_SELECTOR)
+        ) {
+          let blockNumber: Hex | "latest" | undefined = undefined;
+          [, blockNumber] = params;
+
+          const multicallData = decodeFunctionData({
+            abi: multicall3Abi,
+            data: params[0]!.data,
+          });
+          const requests = multicallData.args[0]!.map((call) => ({
+            method: "eth_call",
+            params: [
+              {
+                data: call.callData,
+                to: call.target,
+              },
+              blockNumber,
+            ],
+          })).map((request) =>
+            toLowerCase(JSON.stringify(orderObject(request))),
```

Comment:
It's probably fine, but it's not super intuitive to me that we include the block tag in the hash. Were we doing that before?

Reply:
yes